hack/make/run

```diff
@@ -55,6 +59,29 @@ args="--debug \
 	$storage_params \
 	$extra_params"
 
+if [ -n "$DELVE_PORT" ]; then
+	dockerd="dlv --listen=0.0.0.0:$delve_listen_port \
+		--headless=true \
+		--log \
+		--api-version=2 \
+		--only-same-user=false \
+		--check-go-version=false \
+		--accept-multiclient \
+		exec ${dockerd} --"
+fi
+
 echo "${dockerd} ${args}"
```

Comment:
Some more comments inline :see_no_evil:

```suggestion

dockerd_command="$(command -v dockerd)"
dockerd=( "$dockerd_command" )

if [ -n "$DELVE_PORT" ]; then
	dockerd=(
		dlv
		--listen="0.0.0.0:$delve_listen_port"
		--headless=true
		--log
		--api-version=2
		--only-same-user=false
		--check-go-version=false
		--accept-multiclient
		exec "${dockerd[@]}" --
	)
fi

if [ -n "$DOCKER_ROOTLESS" ]; then
	dockerd=(
		sudo -u "$user" # re-using variables from a conditional that's so far away from this one definitely makes me a little queasy - couldn't those move down here too?
		-E DOCKERD="${dockerd[*]}" # 😬🙈
		-E XDG_RUNTIME_DIR="/tmp/docker-${uid}"
		-E XDG_CONFIG_HOME="/home/${user}/.config"
		-E HOME="/home/${user}"
		--
		dockerd-rootless.sh # any reason this can't just be this, just like it was before?
	)
fi

echo "${dockerd[@]}" "${args}" # this might even be better as just "set -x" before we "exec", frankly (definitely easier to debug)

exec "${dockerd[@]}" ${args}
```

Reply:
Nice idea with these arrays, they're much more readable than a huge string with escaped returns. I refactored handling the dockerd arguments as well.

Regarding the comments:
* `$user` variable - it's not defined in the run script at all, there is nothing to move here
*  `DOCKERD="${dockerd[*]}"` - I know it looks awkward, but if we are to represent commands as arrays instead of strings, it's gonna affect readability when they are being used.
* `dockerd-rootless.sh` - I remember the relative path not working when executed from within the container and since dev container seems to be the first-class citizen for these scripts, I just used the absolute path. I checked it again and now it works, weird. Anyway, thanks for pointing out, I reverted back to relative path. :+1: 
* `set -x` - I agree, much more elegant than echoing the command and its arguments